src/components/hv-picker-field/index.js

```diff
@@ -0,0 +1,247 @@
+import { LOCAL_NAME } from 'hyperview/src/types';
+
+/**
+ * A picker field renders a form field with values that come from a pre-defined list.
+ * - On iOS, pressing the field brings up a bottom sheet with a picker.
+ * - On Android, pressing the field shows the system picker.
+ */
+export default class HvPickerField extends PureComponent<Props, State> {
+  static namespaceURI = Namespaces.HYPERVIEW;
+  static localName = LOCAL_NAME.PICKER_FIELD;
+  props: Props;
+  state: State = {
+    value: null,
+    label: null,
+    pickerValue: null,
+    showPicker: false,
+    fieldPressed: false,
+    savePressed: false,
+    cancelPressed: false,
+  };
+
+  constructor(props: Props) {
+    const { element } = props;
+    super(props);
+    const value = element.getAttribute('value');
+    this.state.value = value;
+    this.state.label = value ? this.getLabelForValue(value) : '';
+
+    this.openPickerModal = this.openPickerModal.bind(this);
+    this.cancelPickerModal = this.cancelPickerModal.bind(this);
+    this.savePickerModal = this.savePickerModal.bind(this);
+  }
+
+  /**
+   * Gets the label from the picker items for the given value.
+   * If the value doesn't have a picker item, returns null.
+   */
+  getLabelForValue = (value: DOMString): ?string => {
+    const { element } = this.props;
+    const elements = element.getElementsByTagNameNS(
+      Namespaces.HYPERVIEW,
+      LOCAL_NAME.PICKER_ITEM,
+    );
+    const item = Array.from(elements).find(
+      e => e.getAttribute('value') === value,
+    );
+    return item ? item.getAttribute('label') : null;
+  };
+
+  /**
+   * Shows the picker, defaulting to the field's value.
+   */
+  openPickerModal = () => {
+    this.setState({
+      pickerValue: this.state.value,
+      showPicker: true,
+    });
+  };
+
+  /**
+   * Hides the picker without applying the chosen value.
+   */
+  cancelPickerModal = () => {
+    this.setState({
+      showPicker: false,
+    });
+  };
+
+  /**
+   * Hides the picker and applies the chosen value to the field.
+   */
+  savePickerModal = () => {
+    const { element } = this.props;
+    this.setState({
+      showPicker: false,
+      value: this.state.pickerValue,
+    });
+    element.setAttribute('value', this.state.pickerValue || '');
+  };
+
+  /**
+   * Renders the picker component. Picker items come from the
+   * <picker-item> elements in the <picker-field> element.
+   */
+  renderPicker = () => {
+    const { element } = this.props;
+    const props = {
+      onValueChange: value => {
+        this.setState({ pickerValue: value });
+      },
+      selectedValue: this.state.pickerValue,
+    };
+
+    // Gets all of the <picker-item> elements. All picker item elements
+    // with a value and label are turned into options for the picker.
+    const children = Array.from(
+      element.getElementsByTagNameNS(
+        Namespaces.HYPERVIEW,
+        LOCAL_NAME.PICKER_ITEM,
+      ),
+    ).map(item => {
+      if (!item) {
```

Comment:
Instead of returning `null` here, perhaps it would be better to `filter` out truthy values before `map`